src/main/java/org/opensearch/neuralsearch/search/query/ScoreDocsMerger.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.query;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.NoArgsConstructor;
+import org.apache.lucene.search.ScoreDoc;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+import static org.opensearch.neuralsearch.search.util.HybridSearchResultFormatUtil.isHybridQueryScoreDocElement;
+
+/**
+ * Merges two ScoreDocs arrays into one
+ */
+@NoArgsConstructor
+public class ScoreDocsMerger {
```

Comment:
nit: HybridQueryMergeOperations

Reply:
I've pulled out merging logic for TopDocs and ScoreDoc into separate classes, this should address this comment and all underlying comment related to naming